lib/danom/monad.rb

```diff
@@ -8,19 +11,65 @@ def initialize(value)
       end
     end
 
+    # Extract the value from the monad. If the underlying value responds to
+    # #value then this will be bypassed instead.
+    #
+    # @see #monad_value
+    # @return [Object] The unwrapped value
     def value
+      if !(Monad === @value) && @value.respond_to?(:value)
+        and_then{ |v| v.value }
```

Comment:
Danom::Monad#value has the variable name 'v'

[Read more about it here.](https://github.com/troessner/reek/blob/master/docs/Uncommunicative-Variable-Name.md)